internal/testutil/acc/atlas.go

```diff
@@ -0,0 +1,116 @@
+package acc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// TestMainExecution must be called from TestMain in the test package if ProjectIDExecution is going to be used.
+func TestMainExecution(m *testing.M) {
+	if !InUnitTest() {
+		atlasInfo.init = true
+		atlasInfo.resourceName = resourceName()
+	}
+
+	exitCode := m.Run()
+
+	if !InUnitTest() && atlasInfo.needsDeletion {
+		fmt.Printf("Deleting execution project: %s, resource: %s\n", atlasInfo.projectName, atlasInfo.resourceName)
+		deleteProject(atlasInfo.projectID)
+		atlasInfo.projectID = ""
```

Comment:
it's not really needed to empty these string as we're about to exit the tests, but it's to make it more clear that those values are invalid from that moment